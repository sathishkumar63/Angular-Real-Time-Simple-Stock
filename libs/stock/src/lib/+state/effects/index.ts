import { StocksEffects } from './stocks.effects';

export const effects: any[] = [StocksEffects];
