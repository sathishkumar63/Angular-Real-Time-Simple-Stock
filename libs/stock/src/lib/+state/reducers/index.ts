import { ActionReducerMap } from '@ngrx/store';
import * as fromStocks from './stocks.reducer';

export * from './stocks.reducer';
export const stocksFeatureKey = 'simpleStocks';

export interface StocksFeatureState {
  stocks: fromStocks.StocksState;
}

export const reducers: ActionReducerMap<StocksFeatureState> = {
  stocks: fromStocks.stocksReducer,
};
