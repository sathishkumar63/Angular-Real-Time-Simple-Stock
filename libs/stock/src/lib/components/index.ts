export * from './stock-card/stock-card.component';
