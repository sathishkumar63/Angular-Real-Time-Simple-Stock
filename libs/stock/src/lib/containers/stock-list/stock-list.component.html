<div class="stock-list">
  <ng-container *ngIf="(stocks$ | async)?.length > 0; else showResiliencyScreen">
    <stock-card
      [stocksList]="stocks$ | async"
      (notifyToggle)="onNotifyToggle($event)"
    ></stock-card>
  </ng-container>
</div>
<ng-template #showResiliencyScreen>
  <stock-error-screen></stock-error-screen>
</ng-template>
