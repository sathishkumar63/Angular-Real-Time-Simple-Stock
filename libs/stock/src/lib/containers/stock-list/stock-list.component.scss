.stock-list {
  background-color: #ffffff;
  padding: 2rem;
  @media only screen and(min-width: 720px) {
    margin: 4rem;
  }
}
