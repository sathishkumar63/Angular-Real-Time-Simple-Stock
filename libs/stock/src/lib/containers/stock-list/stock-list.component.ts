import { Component, OnInit, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'stock-list',
  templateUrl: './stock-list.component.html',
  styleUrls: ['./stock-list.component.scss'],
  encapsulation: ViewEncapsulation.None,
})
export class StockListComponent implements OnInit {
  constructor() {}

  ngOnInit(): void {}
}
