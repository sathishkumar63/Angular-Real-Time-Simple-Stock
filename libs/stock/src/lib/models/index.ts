export * from './stock.constant';
export * from './stock.interface';
export * from './stock.mock.data';
