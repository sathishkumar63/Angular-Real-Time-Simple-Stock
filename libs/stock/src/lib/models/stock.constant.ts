
export const QUOTE_STREAMER_WEBSOCKET = 'wss://streamer.finance.yahoo.com';
export const QUOTE_URL = 'https://query1.finance.yahoo.com/v7/finance/quote?';

export const stocksSymbol: string[] = [
  'AMZN',
  'AAPL',
  'TSLA',
  'GOOG',
  'MSFT'
];
