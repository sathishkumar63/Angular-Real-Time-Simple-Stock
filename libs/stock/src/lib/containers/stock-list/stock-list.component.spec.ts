import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { Store, StoreModule } from '@ngrx/store';
import { of } from 'rxjs';
import * as fromStocks from '../../+state';
import { ErrorScreenComponent, MockStockCardComponent } from '../../components';
import { mockStocks } from '../../models';
import { StockListComponent } from './stock-list.component';

describe('StockListComponent', () => {
  let component: StockListComponent;
  let fixture: ComponentFixture<StockListComponent>;
  let store: Store<fromStocks.StocksFeatureState>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        StockListComponent,
        MockStockCardComponent,
        ErrorScreenComponent,
      ],
      imports: [
        StoreModule.forRoot([]),
        StoreModule.forFeature(
          fromStocks.stocksFeatureKey,
          fromStocks.reducers
        ),
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(StockListComponent);
    component = fixture.componentInstance;
    store = TestBed.inject(Store);
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render resiliency error screen when the stocksList.length < 0', () => {
    // arrange + act
    fixture.detectChanges();
    const stockList = fixture.debugElement.query(By.css('.stock-list'))
      .nativeElement;
    // assert
    expect(stockList).toMatchSnapshot();
  });

  it('should render stocks card when the stocksList.length > 0', () => {
    // arrange
    component.stocksList = mockStocks;
    // act
    fixture.detectChanges();
    const stockList = fixture.debugElement.query(By.css('.stock-list'))
      .nativeElement;
    // assert
    expect(stockList).toMatchSnapshot();
  });

  describe('ngOnInit', () => {
    it('GetStocks action should be dispatched', () => {
      // arrange
      jest.spyOn(store, 'dispatch');
      // act
      component.ngOnInit();
      // assert
      expect(store.dispatch).toHaveBeenCalledWith(new fromStocks.GetStocks());
    });

    it('getStocksList method should be called', () => {
      // arrange
      jest.spyOn(component, 'getStocksList');
      // act
      component.ngOnInit();
      // assert
      expect(component.getStocksList).toBeCalled();
    });
  });

  describe('getStocksList', () => {
    it('should call getStocksData selector', () => {
      // arrange
      const selectSpy = jest.spyOn(store, 'select');
      // act
      component.getStocksList();
      // assert
      expect(selectSpy).toHaveBeenCalledWith(fromStocks.getStocksData);
    });

    it('should set stocksList', () => {
      // arrange
      jest.spyOn(store, 'select').mockImplementation(() => of(mockStocks));
      // act
      component.getStocksList();
      // assert
      expect(component.stocksList).toMatchSnapshot();
    });
  });

  describe('onNotifyToggle', () => {
    it('SetStocks action should be dispatched with stock', () => {
      // arrange
      jest.spyOn(store, 'dispatch');
      const stock = mockStocks[0];
      // act
      component.onNotifyToggle(stock);
      // assert
      expect(store.dispatch).toHaveBeenCalledWith(
        new fromStocks.SetStocks(stock)
      );
    });
  });
});
