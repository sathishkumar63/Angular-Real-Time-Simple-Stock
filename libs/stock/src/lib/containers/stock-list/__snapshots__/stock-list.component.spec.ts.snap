// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockListComponent should render resiliency error screen when the stocks$.length < 0 1`] = `
<div
  class="stock-list"
>
  
  <stock-error-screen>
    <div
      class="stock-error-screen"
    >
      <h2
        class="stock-error-screen__title"
      >
        Uh-oh, something went wrong.
      </h2>
      <p
        class="stock-error-screen__message"
      >
         We are not able to load this page right now. Please refresh this page or follow this 
        <a
          href="https://cors-anywhere.herokuapp.com/corsdemo"
          target="_blank"
        >
          link
        </a>
        . 
      </p>
    </div>
  </stock-error-screen>
</div>
`;

exports[`StockListComponent should render stocks card when the stocks$.length > 0 1`] = `
<div
  class="stock-list"
>
  
  
  <stock-card />
</div>
`;
